components/nexusoperations/executors.go

```diff
@@ -681,31 +664,45 @@ func startCallOutcomeTag(callCtx context.Context, result *nexus.ClientStartOpera
 }
 
 func cancelCallOutcomeTag(callCtx context.Context, callErr error) string {
-	var unexpectedResponseError *nexus.UnexpectedResponseError
+	var handlerErr *nexus.HandlerError
 	if callErr != nil {
 		if callCtx.Err() != nil {
 			return "request-timeout"
 		}
-		if errors.As(callErr, &unexpectedResponseError) {
-			return fmt.Sprintf("request-error:%d", unexpectedResponseError.Response.StatusCode)
+		if errors.As(callErr, &handlerErr) {
+			return fmt.Sprintf("handler-error:%s", handlerErr.Type)
 		}
 		return "unknown-error"
 	}
 	return "successful"
 }
 
-func isRetryableHTTPResponse(response *http.Response) bool {
-	return response.StatusCode >= 500 || slices.Contains(retryable4xxErrorTypes, response.StatusCode)
+func isRetryableHandlerError(eType nexus.HandlerErrorType) bool {
+	switch eType {
+	case nexus.HandlerErrorTypeResourceExhausted,
+		nexus.HandlerErrorTypeInternal,
+		nexus.HandlerErrorTypeUnavailable,
+		nexus.HandlerErrorTypeUpstreamTimeout:
+		return true
+	case nexus.HandlerErrorTypeBadRequest,
+		nexus.HandlerErrorTypeUnauthenticated,
+		nexus.HandlerErrorTypeUnauthorized,
+		nexus.HandlerErrorTypeNotFound,
+		nexus.HandlerErrorTypeNotImplemented:
+		return false
+	default:
+		return true
```

Comment:
```suggestion
		// Default to retryable in case other error types are added in the future.
		// It's better to retry than unexpectedly fail.
		return true
```